pkg/services/query/query_test.go

```diff
@@ -319,6 +319,57 @@ func TestQueryDataMultipleSources(t *testing.T) {
 		assert.Len(t, header.Values("test"), 2)
 	})
 
+	t.Run("forward headers are sent to the datasource", func(t *testing.T) {
+		tc := setup(t)
+		query1, err := simplejson.NewJson([]byte(`
+			{
+				"datasource": {
+					"type": "mysql",
+					"uid": "ds1"
+				},
+				"refId": "A"
+			}
+		`))
+		require.NoError(t, err)
+		query2, err := simplejson.NewJson([]byte(`
+			{
+				"datasource": {
+					"type": "mysql",
+					"uid": "ds2"
+				},
+				"refId": "B"
+			}
+		`))
```

Comment:
why do we need two queries?

Reply:
also, this test seems to be misplaced?
right now, we have it under `TestQueryDataMultipleSources`
let's do `TestQueryDataHeaderForwarding` and then have the barest minimal test that works

Reply:
You are right. Let me move it out. I was squeezing in the test case with an existing test.